### PR TITLE
Misc code cleanup

### PR DESCRIPTION
**Pull request by stelfrag**
##### Summary
Code cleanup / refactoring for health and ACLK

- Change queries to bind parameters if possible instead of using snprintfz to build the query
- Reformat queries for readability
- Use macros for better readability when binding / fetching parameters 
- Use one cleanup function for claimed / not claimed agents
- Remove unused code / macros
- Adjust an index on health_log_detail

### CODE REVIEW

**Comment by thiagoftsm** [comment]
Hello @stelfrag ,

Do you plan to address [this](https://app.codacy.com/gh/netdata/netdata/pullRequest?prid=12320970) codacy warning?

Best regards!

**Comment by stelfrag** [comment]
> Hello @stelfrag ,
> 
> Do you plan to address [this](https://app.codacy.com/gh/netdata/netdata/pullRequest?prid=12320970) codacy warning?
> 
> Best regards!

Tried to "fool" it with no success :smile: so lets ignore it

File: database/sqlite/sqlite_health.c [hunk 1]
```diff
@@ -362,46 +378,67 @@ void sql_health_alarm_log_save(RRDHOST *host, ALARM_ENTRY *ae)
 /* Health related SQL queries
    Get a count of rows from health log table
 */
-#define SQL_COUNT_HEALTH_LOG_DETAIL "SELECT count(1) FROM health_log_detail hld, health_log hl where hl.host_id = @host_id and hl.health_log_id = hld.health_log_id;"
-void sql_health_alarm_log_count(RRDHOST *host) {
+#define SQL_COUNT_HEALTH_LOG_DETAIL "SELECT count(1) FROM health_log_detail hld, health_log hl " \
```
Comment (thiagoftsm) [inline database/sqlite/sqlite_health.c:381]: I did not get the necessity to use the function `count`, why are we not using only `SELECT 1 FROM ... `? :thinking:

File: database/sqlite/sqlite_health.c [hunk 2]
```diff
@@ -362,46 +378,67 @@ void sql_health_alarm_log_save(RRDHOST *host, ALARM_ENTRY *ae)
...
-        return;
+        goto fail;
     }
 
     rc = sqlite3_step_monitored(res);
     if (likely(rc == SQLITE_ROW))
-        host->health.health_log_entries_written = (size_t) sqlite3_column_int64(res, 0);
+        entries_in_db = (int) sqlite3_column_int64(res, 0);
 
+fail:
```
Comment (thiagoftsm) [inline database/sqlite/sqlite_health.c:412]: we are going to `fail` when we also have success, probably we need another name. :)

File: database/sqlite/sqlite_health.c [hunk 3]
```diff
@@ -414,137 +451,63 @@ void sql_health_alarm_log_cleanup_not_claimed(RRDHOST *host) {
...
         error_report("Failed to cleanup health log detail table, rc = %d", rc);
 
+    int rows = sql_health_alarm_log_count(host);
+    if (rows >= 0)
+        host->health.health_log_entries_written = rows;
+
+    if (aclk_table_exists)
+        sql_aclk_alert_clean_dead_entries(host);
+
+fail:
```
Comment (thiagoftsm) [inline database/sqlite/sqlite_health.c:494]: the same suggestion above for name.

Comment (stelfrag) [reply database/sqlite/sqlite_health.c:494]: good point, will change to `done`

File: database/sqlite/sqlite_health.c (hunk 2 above)
Comment (stelfrag) [reply database/sqlite/sqlite_health.c:412]: good point, will change to `done`

File: database/sqlite/sqlite_health.c (hunk 1 above)
Comment (stelfrag) [reply database/sqlite/sqlite_health.c:381]: Need the number of entries as in select count(*), not just 1

File: database/sqlite/sqlite_health.c [hunk 4]
```diff
@@ -1282,16 +1216,17 @@ int alert_hash_and_store_config(
     return 1;
 }
 
-#define SQL_SELECT_HEALTH_LAST_EXECUTED_EVENT "SELECT hld.new_status FROM health_log hl, health_log_detail hld WHERE hl.alarm_id = %u AND hld.unique_id != %u AND hld.flags & %u AND hl.host_id = @host_id and hl.health_log_id = hld.health_log_id ORDER BY hld.unique_id DESC LIMIT 1;"
+#define SQL_SELECT_HEALTH_LAST_EXECUTED_EVENT                                                                          \
+    "SELECT hld.new_status FROM health_log hl, health_log_detail hld "                                                 \
+    "WHERE hl.host_id = @host_id AND hl.alarm_id = @alarm_id AND hld.unique_id != @unique_id AND hld.flags & @flags "  \
+    "AND hl.health_log_id = hld.health_log_id ORDER BY hld.unique_id DESC LIMIT 1;"
```
Comment (thiagoftsm) [inline database/sqlite/sqlite_health.c:1222]: Considering we are still using `unique_id` as Unix epoch, I agree 100% that we need to take a look on them in `DESC` order, but we need to bring another change for this query to make more sense. We are asking in `DESC` for an index created in [ASC](https://github.com/netdata/netdata/blob/3c78c045bb0ea7225b35df4b6a83e671fc68aed1/database/sqlite/sqlite_functions.c#L58) order. Please, modify the index.

File: database/sqlite/sqlite_health.c [hunk 5]
```diff
@@ -1747,42 +1726,42 @@ void sql_update_alarm_with_config_hash(RRDHOST *host, uint32_t alarm_id, uint64_
...
-    if (unlikely(rc != SQLITE_DONE)) {
+    if (unlikely(rc != SQLITE_DONE))
         error_report("Failed to execute SQL_UPDATE_ALARM_ID_WITH_CONFIG_HASH, rc = %d", rc);
-        rc = sqlite3_finalize(res);
-        if (unlikely(rc != SQLITE_OK))
-            error_report("Failed to reset statement to update health log detail table with config hash ids, rc = %d", rc);
-        return;
-    }
+
+fail:
```
Comment (thiagoftsm) [inline database/sqlite/sqlite_health.c:1754]: The same suggestion about name here.

Comment (stelfrag) [reply database/sqlite/sqlite_health.c:1754]: good point, will change to done as well

**Review by thiagoftsm — DISMISSED** [review]
After last commits I do not have more nothing to ask. Coverity did not report nothing and everything is working as expected, LGTM!

File: database/sqlite/sqlite_aclk_alert.c [hunk 6]
```diff
@@ -7,43 +7,58 @@
...
+    rc = sqlite3_prepare_v2(db_meta, SQL_SELECT_VARIABLE_ALERT_BY_UNIQUE_ID, -1, &res, 0);
     if (rc != SQLITE_OK) {
         error_report("Failed to prepare statement when trying to check for alert variables.");
         return false;
     }
 
-    rc = sqlite3_bind_blob(res, 1, host_id, sizeof(*host_id), SQLITE_STATIC);
+    rc = sqlite3_bind_int(res, 1, (int) unique_id);
     if (unlikely(rc != SQLITE_OK)) {
         error_report("Failed to bind host_id for checking alert variable.");
```
Comment (MrZammler) [inline database/sqlite/sqlite_aclk_alert.c:47]:
```suggestion
        error_report("Failed to bind unique_id for checking alert variable.");
```

File: database/sqlite/sqlite_aclk_alert.c [hunk 7]
```diff
@@ -54,20 +69,22 @@ static inline bool is_event_from_alert_variable_config(uint32_t unique_id, uuid_
...
+
+static bool should_send_to_cloud(RRDHOST *host, ALARM_ENTRY *ae)
 {
     sqlite3_stmt *res = NULL;
     char uuid_str[UUID_STR_LEN];
     uuid_unparse_lower_fix(&host->host_uuid, uuid_str);
-    int send = 1;
 
-    if (ae->new_status == RRDCALC_STATUS_REMOVED || ae->new_status == RRDCALC_STATUS_UNINITIALIZED) {
+    bool send = false;
```
Comment (MrZammler) [inline database/sqlite/sqlite_aclk_alert.c:84]: Should we keep the initial status to `true` ? If for any reason the binds later fail and it goes to done, we should send it just to be safe.. ?

File: database/sqlite/sqlite_aclk_alert.c [hunk 8]
```diff
@@ -564,37 +583,35 @@ int aclk_push_alert_config_event(char *node_id __maybe_unused, char *config_hash
...
-        alarm_config.recipient = sqlite3_column_bytes(res, 20) > 0 ? strdupz((char *)sqlite3_column_text(res, 20)) : NULL;
-        alarm_config.exec = sqlite3_column_bytes(res, 21) > 0 ? strdupz((char *)sqlite3_column_text(res, 21)) : NULL;
-        alarm_config.delay = sqlite3_column_bytes(res, 22) > 0 ? strdupz((char *)sqlite3_column_text(res, 22)) : NULL;
-        alarm_config.repeat = sqlite3_column_bytes(res, 23) > 0 ? strdupz((char *)sqlite3_column_text(res, 23)) : NULL;
-        alarm_config.info = sqlite3_column_bytes(res, 24) > 0 ? strdupz((char *)sqlite3_column_text(res, 24)) : NULL;
-        alarm_config.options = sqlite3_column_bytes(res, 25) > 0 ? strdupz((char *)sqlite3_column_text(res, 25)) : NULL;
-        alarm_config.host_labels = sqlite3_column_bytes(res, 26) > 0 ? strdupz((char *)sqlite3_column_text(res, 26)) : NULL;
+        int param = 0;
+        alarm_config.alarm = SQLITE3_COLUMN_STRDUPZ_OR_NULL(res, param++);
+        alarm_config.alarm = SQLITE3_COLUMN_STRDUPZ_OR_NULL(res, param++);
```
Comment (MrZammler) [inline database/sqlite/sqlite_aclk_alert.c:588]: Stelio, `alarm` is bound twice?

File: database/sqlite/sqlite_aclk_alert.c (hunk 7 above)
Comment (stelfrag) [reply database/sqlite/sqlite_aclk_alert.c:84]: this is to be on the safe side not to send excessive messages to the cloud if things fail locally

File: database/sqlite/sqlite_aclk_alert.c (hunk 8 above)
Comment (stelfrag) [reply database/sqlite/sqlite_aclk_alert.c:588]: It's wrong I will push a commit

**Review by MrZammler — APPROVED** [review]
Looks good!

**Review by thiagoftsm — APPROVED** [review]
PR is working as expected, LGTM!

**Pull request merged**